astronomer/providers/core/sensors/astro.py

```diff
@@ -0,0 +1,115 @@
+
+from astronomer.providers.core.hooks.astro import AstroHook
+from astronomer.providers.core.triggers.astro import AstroDeploymentTrigger
+from astronomer.providers.utils.typing_compat import Context
+
+
+class ExternalDeploymentSensor(BaseSensorOperator):
+    """
+    Custom Apache Airflow sensor for monitoring external deployments using Astro Cloud.
+
+    :param external_dag_id: External ID of the DAG being monitored.
+    :param astro_cloud_conn_id: The connection ID to retrieve Astro Cloud credentials.
+                               Defaults to "astro_cloud_default".
+    :param external_task_id: External ID of the task being monitored. If None, monitors the entire DAG.
+    :param kwargs: Additional keyword arguments passed to the BaseSensorOperator constructor.
+    """
+
+    def __init__(
+        self,
+        external_dag_id: str,
+        astro_cloud_conn_id: str = "astro_cloud_default",
+        external_task_id: str | None = None,
+        **kwargs: Any,
+    ):
+        super().__init__(**kwargs)
+        self.astro_cloud_conn_id = astro_cloud_conn_id
+        self.external_task_id = external_task_id
+        self.external_dag_id = external_dag_id
+        self._dag_run_id: str = ""
+
+    # def wait_for_dag_start(self, second_to_wait: int = 120, sleep: int = 5) -> None:
+    #     """TODO"""
+    #     hook = AstroHook(self.astro_cloud_conn_id)
+    #     end_time = datetime.datetime.now() + datetime.timedelta(seconds=second_to_wait)
+    #     while end_time >= datetime.datetime.now():
+    #         try:
+    #             dag_runs = hook.get_dag_runs(self.external_dag_id)
+    #             if dag_runs is not None:
+    #                 return
+    #         except Exception:
+    #             time.sleep(sleep)
+
+    def poke(self, context: Context) -> bool | PokeReturnValue:
+        """
+        Method to check if the external deployment is successful.
+
+        Queries Astro Cloud for the status of the specified DAG or task instance.
+        Returns True if successful, False otherwise.
+
+        :param context: The task execution context.
+        """
+        hook = AstroHook(self.astro_cloud_conn_id)
+        dag_runs: list[dict[str, Any]] = hook.get_dag_runs(self.external_dag_id)
+        if dag_runs is None or len(dag_runs) == 0:
+            self.log.info(f"No dag-runs found for dag-id {self.external_dag_id}")
+            return True
+        self._dag_run_id = cast(str, dag_runs[0]["dag_run_id"])
+        if self.external_task_id is not None:
+            task_instance = hook.get_task_instance(
+                self.external_dag_id, self._dag_run_id, self.external_task_id
+            )
+            task_state = task_instance.get("state") if task_instance else None
+            if task_state == "success":
+                return True
+        else:
+            state = dag_runs[0].get("state")
+            if state == "success":
+                return True
+        return False
```

Comment:
Do I understand correctly this fetches the latest DAG run and checks the status?

Reply:
yes, it check the status of last dagrun based on execution date